pkg/prometheus/prometheus_test.go

```diff
@@ -0,0 +1,633 @@
+						Category: "snapshot",
+						Gadget:   "stubsnapshotter",
+					},
+				},
+			},
+			expectedInt64Gauges: map[string]map[string]int64{
+				"gauge_no_labels_nor_filtering": {"": 5},
+			},
+		},
+		{
+			name: "gauge_filter_only_root_events",
+			config: &Config{
+				MetricsName: "gauge_filter_only_root_events",
+				Metrics: []Metric{
+					{
+						Name:     "gauge_filter_only_root_events",
+						Type:     "gauge",
+						Category: "snapshot",
+						Gadget:   "stubsnapshotter",
+						Selector: []string{"uid:0"},
+					},
+				},
+			},
+			expectedInt64Gauges: map[string]map[string]int64{
+				"gauge_filter_only_root_events": {"": 3},
+			},
+		},
+		{
+			name: "gauge_filter_only_root_cat_events",
+			config: &Config{
+				MetricsName: "gauge_filter_only_root_cat_events",
+				Metrics: []Metric{
+					{
+						Name:     "gauge_filter_only_root_cat_events",
+						Type:     "gauge",
+						Category: "snapshot",
+						Gadget:   "stubsnapshotter",
+						Selector: []string{"uid:0", "comm:cat"},
+					},
+				},
+			},
+			expectedInt64Gauges: map[string]map[string]int64{
+				"gauge_filter_only_root_cat_events": {"": 2},
+			},
+		},
+		{
+			name: "gauge_with_int_field",
+			config: &Config{
+				MetricsName: "gauge_with_int_field",
+				Metrics: []Metric{
+					{
+						Name:     "gauge_with_int_field",
+						Type:     "gauge",
+						Category: "snapshot",
+						Gadget:   "stubsnapshotter",
+						Field:    "intval",
+					},
+				},
+			},
+			expectedInt64Gauges: map[string]map[string]int64{
+				"gauge_with_int_field": {"": 105 + 216 + 327 + 428 + 429},
+			},
+		},
+		{
+			name: "gauge_with_float_field",
+			config: &Config{
+				MetricsName: "gauge_with_float_field",
+				Metrics: []Metric{
+					{
+						Name:     "gauge_with_float_field",
+						Type:     "gauge",
+						Category: "snapshot",
+						Gadget:   "stubsnapshotter",
+						Field:    "floatval",
+					},
+				},
+			},
+			expectedFloat64Gauges: map[string]map[string]float64{
+				"gauge_with_float_field": {"": 201.2 + 423.3 + 645.4 + 867.5 + 1089.6},
+			},
+		},
+		{
+			name: "gauge_multiple",
+			config: &Config{
+				MetricsName: "gauge_multiple",
+				Metrics: []Metric{
+					{
+						Name:     "gauge_no_labels_nor_filtering",
+						Type:     "gauge",
+						Category: "snapshot",
+						Gadget:   "stubsnapshotter",
+					},
+					{
+						Name:     "gauge_filter_only_root_events",
+						Type:     "gauge",
+						Category: "snapshot",
+						Gadget:   "stubsnapshotter",
+						Selector: []string{"uid:0"},
+					},
+				},
+			},
+			expectedInt64Gauges: map[string]map[string]int64{
+				"gauge_no_labels_nor_filtering": {"": 5},
+				"gauge_filter_only_root_events": {"": 3},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			t.Cleanup(cancel)
+
+			wg := &sync.WaitGroup{}
+			wg.Add(len(test.config.Metrics))
```

Comment:
The test fails after the 10-minute timeout for me.

@mauriciovasquezbernal @flyth Who is calling `wg.Done()` for metrics of type `nonvalidtype` (used in TestMetrics)?

Reply:
For those cases the code doesn't arrive to this check but returns early in 
https://github.com/inspektor-gadget/inspektor-gadget/blob/787ecf3e17a9404a0cfa52003abc6bc9faa134d8/pkg/prometheus/prometheus_test.go#L574-L577

Reply:
Do you have more details about your issues / environment? I checked and they work fine in my machine:

```bash
$ cd pkg/prometheus
$ go test  ./... -count 1
ok  	github.com/inspektor-gadget/inspektor-gadget/pkg/prometheus	0.004s
```

Reply:
It was during my tests in #1809.

When pkg/runtime/local/local.go RunGadget() returns an error (because of my local changes), then it times out.

Reply:
Unfortunately it's not easy to handle as RunGadget() is called in a different goroutine https://github.com/inspektor-gadget/inspektor-gadget/blob/787ecf3e17a9404a0cfa52003abc6bc9faa134d8/pkg/prometheus/prometheus.go#L272 and its return error is ignored. 

Perhaps a solution is to implementing something like https://stackoverflow.com/questions/32840687/timeout-for-waitgroup-wait to avoid waiting for the whole test suite to timeout. WDYT?

Reply:
Yes, the timeout solution in the test sounds fine.

Reply:
Opened https://github.com/inspektor-gadget/inspektor-gadget/pull/1833.